### PR TITLE
core: add wal/db size flag, enable limit: for wal/db devices

### PR DESCRIPTION
**Pull request by jschmid1**
$title

-----------------

**Checklist:**
- [x] Added unittests and or functional tests
- [x] Adapted documentation ~#TODO~ adapted the wiki
~- [ ] Referenced issues or internal bugtracker~
- [x] Ran integration tests successfully (trigger with "@susebot run teuthology" in a GitHub comment; see the [wiki](https://github.com/SUSE/DeepSea/wiki/Testing) for more information)

### CODE REVIEW

**Comment by jschmid1** [comment]
@susebot run teuthology

File: srv/salt/_modules/dg.py [hunk 1]
```diff
@@ -590,12 +590,16 @@ def wal_device_attrs(self) -> dict:
         return self.filter_args.get("wal_devices", dict())
 
     @property
-    def limit(self) -> int:
-        """ Limits the amount of devices assigned
+    def block_wal_size(self) -> dict:
```
Comment (tserong) [inline srv/salt/_modules/dg.py]: Does this really return a dict?

File: srv/salt/_modules/dg.py [hunk 2]
```diff
@@ -590,12 +590,16 @@ def wal_device_attrs(self) -> dict:
...
+        """ Wal Device attributes
+        """
+        return self.filter_args.get("block_wal_size", False)
 
-        Limit 0 -> unlimited
+    @property
+    def block_db_size(self) -> dict:
```
Comment (tserong) [inline srv/salt/_modules/dg.py]: Likewise?

File: srv/salt/_modules/dg.py (hunk 1 above)
Comment (jschmid1) [reply srv/salt/_modules/dg.py]: will fix soonish

File: srv/salt/_modules/dg.py (hunk 2 above)
Comment (jschmid1) [reply srv/salt/_modules/dg.py]: will fix soonish

**Comment by jschmid1** [comment]
review comments addressed and rebased

**Comment by jschmid1** [comment]
@susebot run teuthology

**Pull request merged**